pkg/pprof/fix_go_truncated.go

```diff
@@ -23,11 +23,20 @@ const (
 // MayHaveGoHeapTruncatedStacktraces reports whether there are
 // any chances that the profile may have truncated stack traces.
 func MayHaveGoHeapTruncatedStacktraces(p *profilev1.Profile) bool {
-	if !hasGoHeapSampleTypes(p) {
+	var minDepth int
+
+	if hasGoHeapSampleTypes(p) {
+		// TODO: Go heap profiles in Go 1.23 have a depth limit of 128 frames. At some point we can probably increase/remove this
+		minDepth = 32
+	} else if hasGoCPUSampleTypes(p) {
+		minDepth = 64
+	} else {
```

Comment:
Should we do it now? :) The function could report `false`, if the profile contains at least one stack trace strictly longer/taller that the limit by some margin (not accounting for skipped frames, probably – not sure). Like, we do not want to try to repair a heap profile, if it has a stack trace with 33 frames

WDYT?